src/Http/Http/test/RequestCookiesCollectionTests.cs

```diff
@@ -38,5 +38,13 @@ public void AppContextSwitchUnEscapesKeysAndValues(string input, string expected
             Assert.Equal(expectedKey, cookies.Keys.Single());
             Assert.Equal(expectedValue, cookies[expectedKey]);
         }
+
+        [Fact]
+        public void ParseManyCookies()
+        {
+            var cookies = RequestCookieCollection.Parse(new StringValues(new[] { "a=a", "b=b", "c=c", "d=d", "e=e", "f=f", "g=g", "h=h", "i=i", "j=j", "k=k", "l=l" }));
+
+            Assert.Equal(12, cookies.Count);
```

Comment:
Was this failing before this change?

Reply:
Maybe we should also add a crazy big variant of this that parses 1000 cookies as well?

Reply:
This test did trigger the Debug.Assert inside Parse. The Count Assert was just to make sure items weren't being lost.

Reply:
> Maybe we should also add a crazy big variant of this that parses 1000 cookies as well?

10 is the only boundary condition in todays logic, anything bigger than that ends up testing Dictionary.